### ISSUE TITLE
运行环境

### ISSUE THREAD

**Issue by wuliebucha** [chineseocr/chineseocr#130]
问题：训练时loss和acc总是为0；
原因：计算 cost = criterion(preds, text, preds_size, length) / batch_size 时，如果preds在gpu上，则loss为0，preds 在 cpu上，则loss正常。
我的conda运行环境：pytorch-0.4.1、cudatoolkit-9.0、cudnn-7.1.2、warp-ctc为https://github.com/SeanNaren/warp-ctc，怀疑可能是因为环境问题，导致loss为0，不知大家的环境是什么样的？为什么preds在gpu上loss为0？非常感谢

**Comment by wuliebucha** [comment]
`import numpy as np
import torch
from torch import Tensor
from torch.autograd import Variable
from warpctc_pytorch import CTCLoss
import os
import torch.nn as nn
# torch_ctc_loss = nn.CTCLoss(blank=0)
os.environ["CUDA_VISIBLE_DEVICES"] = "3"

criterion = CTCLoss()
batch_size = 16
seqLength = 71
outputDim = 5992
out_seq_len = 10
acts = Variable(torch.randn((seqLength, batch_size, outputDim)), requires_grad=True)
acts1 = Variable(torch.randn((seqLength, batch_size, outputDim)), requires_grad=True).cuda()
labels = Variable(torch.from_numpy(np.random.randint(1, outputDim, (batch_size, out_seq_len))).int()).view(-1)
act_lens = Variable(Tensor([seqLength] * batch_size).int())
label_lens = Variable(Tensor([out_seq_len] * batch_size).int())

loss = criterion(acts, labels, act_lens, label_lens)
loss1 = criterion(acts1, labels, act_lens, label_lens)

print("acts size: {}, dtype:{}, data_range:({},{})".format(acts.size(), acts.dtype, torch.min(acts.data),
                                                           torch.max(acts.data)))
print("labels size: {}, dtype:{},  data_range:({},{})".format(labels.size(), labels.dtype, torch.min(labels.data),
                                                              torch.max(labels.data)))
print(
    "acts_len size: {}, dtype:{},  data_range:({},{})".format(act_lens.size(), act_lens.dtype, torch.min(act_lens.data),
                                                              torch.max(act_lens.data)))
print("labels_len size: {}, dtype:{},  data_range:({},{})".format(label_lens.size(), label_lens.dtype,
                                                                  torch.min(label_lens.data),
                                                                  torch.max(label_lens.data)))

print("loss: {}".format(loss.data[0]))
print("loss1: {}".format(loss1.data[0]))`
output:
acts size: torch.Size([71, 16, 5992]), dtype:torch.float32, data_range:(-5.402641773223877,5.229351997375488)
labels size: torch.Size([160]), dtype:torch.int32,  data_range:(11,5987)
acts_len size: torch.Size([16]), dtype:torch.int32,  data_range:(71,71)
labels_len size: torch.Size([16]), dtype:torch.int32,  data_range:(10,10)
loss: 9456.689453125
loss1: 0.0
